### ISSUE TITLE
Error when mirroring data to BigQuery TypeError: Converting circular structure to JSON

### ISSUE THREAD

**Issue by MrAlek** [firebase/extensions#236]
### [REQUIRED] Step 2: Describe your configuration

- Extension name: **`firestore-bigquery-export`
- Extension version: **0.1.2**

### [REQUIRED] Step 3: Describe the problem

After running the initial setup and upon changing a field on a document in Firestore, the `ext-firestore-bigquery-export-fsexportbigquery` ran with the following error:

```
Error when mirroring data to BigQuery TypeError: Converting circular structure to JSON
    at JSON.stringify (<anonymous>)
    at events.map (/srv/node_modules/@firebaseextensions/firestore-bigquery-change-tracker/lib/bigquery/index.js:64:36)
    at Array.map (<anonymous>)
    at FirestoreBigQueryEventHistoryTracker.<anonymous> (/srv/node_modules/@firebaseextensions/firestore-bigquery-change-tracker/lib/bigquery/index.js:56:33)
    at Generator.next (<anonymous>)
    at fulfilled (/srv/node_modules/@firebaseextensions/firestore-bigquery-change-tracker/lib/bigquery/index.js:20:58)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:229:7)
```

This is the document for which the error occured:

<img width="457" alt="Skärmavbild 2020-03-22 kl  22 07 55" src="https://user-images.githubusercontent.com/1496135/77260862-cbe09080-6c8a-11ea-97c3-f1ef2bfd8e47.png">

**Comment by russellwheatley** [comment]
Hi @MrAlek, this is an interesting error. I've tried to recreate the circular reference dependency error and I was unable to reproduce. Could you please confirm what type of change you made to the document? Is it in a collection or a sub-collection?

**Comment by MrAlek** [comment]
@russellwheatley In this case I just changed one of the string attributes to test the integration. It's in a root-level collection but as you can see the document also has sub-collections.

**Comment by MrAlek** [comment]
Here's the full log:
<img width="1023" alt="image" src="https://user-images.githubusercontent.com/1496135/77323418-83ba8000-6d15-11ea-892c-440d02d1a599.png">

**Comment by handihow** [comment]
same error here..
Just installed this extension but I have the same error when updating documents in the collection

**Comment by ThadeuLuz** [comment]
Also getting this. Will investigate better tomorrow

**Comment by mikekaranja** [comment]
<img width="1030" alt="Screenshot 2020-03-30 at 01 48 41" src="https://user-images.githubusercontent.com/12885216/77863141-b63e0e80-7228-11ea-914a-6104e0b723fb.png">
I have the exact same issue. Any timelines on when it will be fixed?

**Comment by rafaelacb** [comment]
I am with this issue too.. I've just installed BigQuery extension on my firebase project and made a test with a bigquery-mirror-test and I've this... Is there any way to solve it quickly?

**Comment by ripcity666** [comment]
oh I have no idea. I'm trying to figure out how to get rid of my psycho
neighbor. she put all kinds of shit on my phone

**Comment by fguespe** [comment]
You can follow up here
https://github.com/firebase/extensions/issues/235

**Comment by laurenzlong** [comment]
The fix has been published in 0.1.3 of the [firestore-bigquery-change-tracker NPM package](https://www.npmjs.com/package/@firebaseextensions/firestore-bigquery-change-tracker) - which is a dependency for the extension. When we do the next release of the firestore-bigquery-export extension, you'll receive the update. If you want to "force" the dependency update now, you can either:
- uninstall and reinstall the extension
- OR [reconfigure the extension](https://firebase.google.com/docs/extensions/manage-installed-extensions?platform=console#reconfigure) to force an update

**Comment by laurenzlong** [comment]
The fix is now live in v0.1.4 of the extension, you can update your extension by following the instructions in https://firebase.google.com/docs/extensions/manage-installed-extensions?platform=console#update-version. Please create a new issue if you still experience problems. Thanks for filing!
